contracts/zero-ex/contracts/src/ZeroEx.sol

```diff
@@ -0,0 +1,122 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-utils/contracts/src/v06/LibBytesV06.sol";
+import "./interfaces/IZeroExBootstrapper.sol";
+import "./storage/LibProxyStorage.sol";
+import "./errors/LibProxyRichErrors.sol";
+
+
+/// @dev An extensible proxy contract that serves as a universal entry point for
+///      interacting with the 0x protocol.
+contract ZeroEx {
+
+    // solhint-disable separate-by-one-line-in-contract,indent,var-name-mixedcase
+
+    using LibBytesV06 for bytes;
+
+    /// @dev Construct this contract.
+    ///      After constructing this contract, the deployer should call
+    ///      `bootstrap()` to seed the initial feature set.
+    constructor() public {
+        // Set the `bootstrap()` caller to the deployer.
+        LibProxyStorage.getStorage().bootstrapCaller = msg.sender;
+    }
+
+    // solhint-disable state-visibility
+
+    /// @dev Forwards calls to the appropriate implementation contract.
+    fallback() external payable {
+        bytes4 selector = msg.data.readBytes4(0);
```

Comment:
Solidity supports native calldata array slices now FYI: https://solidity.readthedocs.io/en/v0.6.0/types.html#array-slices. I think that means we can get rid of `LibBytesV06`?

Reply:
Best news I've heard all month.

Reply:
> As of now, array slices are only implemented for calldata arrays.

womp, womp

Reply:
Still can't cast to `bytes4` without assembly and we'll probably use LibBytes sooner than we think so I think I'll keep it for now.